app/logic/volunteers.py

```diff
@@ -19,45 +21,67 @@ def getEventLengthInHours(startTime, endTime, eventDate):
 
 def updateVolunteers(participantData):
     """
-    updates the events participant table with data from the track Volunteer webpage
+    Create new entry in event participant table if user does not exist. Otherwise, updates the record.
 
     param: participantData- a dictionary that contains data from every row of the page along with the associated username.
     """
-
     for user in range(1, len(participantData)):
         if f'username{user}' in participantData:
             username = participantData[f'username{user}']
-            if (User.get_or_none(User.username == username)):
-
+            userObject = User.get_or_none(User.username==username)
+            event = Event.get_or_none(Event.id==participantData['event'])
+            eventParticipant = EventParticipant.get_or_none(user=userObject, event=participantData['event'])
+            if userObject:
                 try:
-                    if participantData['checkbox_'+ username] == "on": #if the user is marked as present
-                        (EventParticipant.update({EventParticipant.hoursEarned: float(participantData['inputHours_'+ username]),EventParticipant.attended: True}).where(EventParticipant.event == participantData['event'],
-                                                 EventParticipant.user == participantData[f'username{user}'])).execute()
-
+                    if participantData[f'checkbox_{username}']: #if the user is marked as present
+                        hoursEarned = float(participantData['inputHours_'+ username])
+                        if eventParticipant:
+                            ((EventParticipant
+                                .update({
+                                    EventParticipant.hoursEarned: hoursEarned})
```

Comment:
Where to line-break is often a personal preference thing, but I think this one makes it harder to read. I'd keep the dictionary all on one line, since there is only one key:value pair